tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c

```diff
@@ -133,3 +133,37 @@ void test_mheap_block_desc(void)
 		k_free(block[i]);
 	}
 }
+
+void test_mheap_block_release(void)
+{
+	void *block[BLK_NUM_MAX], *block_fail;
+
+	if (IS_ENABLED(CONFIG_MEM_POOL_HEAP_BACKEND)) {
+		ztest_test_skip();
+	}
+
+	/**
+	 * TESTPOINT: When the blocks in the heap memory pool are free by
+	 * the function k_free, the region would be released back to the
+	 * heap memory pool.
+	 */
+	for (int i = 0; i < BLK_NUM_MAX; i++) {
```

Comment:
This is fragile.  The **old** mem_pool had an architecture where you could know exactly the alignment and layout of the sub-blocks and thus compute an easy "maximum blocks" number.  The new one is more fragmentation resistant and defeats that.  I had to do a lot of tuning to get the compatibility API to produce the same results as the old code for these tests (and @npitre can confirm that's a huge pain, as he had to repeat the process when optimizing the memory usage!).

If you want to know that a heap is full, you need to keep allocating until it fails and not just assume you know the right maximum block count.  A few tests here (or maybe in another test case?) were changed to work like that, you can copy code from there if you like.

Reply:
Thanks, I found the way in test_mheap_malloc_free().